brainglobe_stitch/tile.py

```diff
@@ -55,3 +55,148 @@ def __init__(
         self.tile_id: int = int(attributes["tile"])
         self.illumination_id: int = int(attributes["illumination"])
         self.angle: float = float(attributes["angle"])
+
+
+class Overlap:
+    """
+    Overlap class to store information about the overlap between two tiles.
+
+    Attributes
+    ----------
+    coordinates : npt.NDArray
+        The coordinates for the start of the overlap in the fused image.
+    size : List[npt.NDArray]
+        The size of the overlap in the fused image for each resolution pyramid.
+    tiles : Tuple[Tile, Tile]
+        The overlapping tiles.
+    local_coordinates : List[Tuple[npt.NDArray, npt.NDArray]]
+        A list of tuples representing the starting coordinates for the overlap
+        in each tile's data pyramid for each resolution pyramid level.
+        The tuple contains the starting coordinates for the overlap in each
+        tile.
+    """
+
+    def __init__(
+        self,
+        overlap_coordinates: npt.NDArray,
+        overlap_size: npt.NDArray,
+        tile_i: Tile,
+        tile_j: Tile,
+    ):
+        """
+        Initialize the overlap.
+
+        Parameters
+        ----------
+        overlap_coordinates: npt.NDArray
+            The starting coordinates for the overlap in the fused image.
+        overlap_size: npt.NDArray
+            The size of the overlap in the fused image [z,y,x].
+        tile_i: Tile
+            The first tile.
+        tile_j: Tile
+            The second tile.
+        """
+        self.coordinates: npt.NDArray = overlap_coordinates
```

Comment:
Oh nice didn't know about `numpy.typing.NDArray`... I was just using `np.ndarray`

Reply:
It's very handy! You can also specify the specific type the array is holding.